src/backend/cgcod.c

```diff
@@ -726,7 +726,7 @@ code *prolog()
     assert(usedalloca != 1);
     AllocaOff = alignsection(usedalloca ? (Foff - REGSIZE) : Foff, REGSIZE, bias);
 
-    CSoff = alignsection(AllocaOff - cstop * REGSIZE, REGSIZE, bias);
+    CSoff = alignsection(AllocaOff - (int)cstop * REGSIZE, REGSIZE, bias);
```

Comment:
BTW: the bad codegen that compensates the problem for dmc compiled dmd is that the negation of a long long in EDX:EAX is reordered:
```
		neg	EAX
		neg	EDX
		sbb	EDX,0
```
in the optimizing build (with dmc 8.57), while it's
```
		neg	EDX
		neg	EAX
		sbb	EDX,0
```
in the debug build. The optimized version happens to work, because the result is expected to be positive and the wrong instruction order alway clears to high word.